www/outsystems-secure-sqlite-init.js

```diff
@@ -33,31 +61,52 @@ function acquireLsk(successCallback, errorCallback) {
     var initFn = function() {
         var ss = new SecureStorage(
             function () {
-                // and when initialized attempt to get the stored key
-                ss.get(
-                    function (value) {
-                        lskCache = value;
-                        console.log("Got Local Storage key");
-                        successCallback(lskCache);
-                    },
-                    function (error) {
-                        // If there's no key yet, generate a new one and store it
-                        var newKey = generateKey();
-                        lskCache = undefined;
-                        console.log("Setting new Local Storage key");
-                        ss.set(
-                            function (key) {
-                                lskCache = newKey;
-                                successCallback(lskCache);
-                            },
-                            errorCallback,
-                            LOCAL_STORAGE_KEY,
-                            newKey);
-                    },
-                    LOCAL_STORAGE_KEY);
+                rewriteLsk(ss, function(){
+                    // Now, lets try to get all keys from OutSystems keystore
+                    ss.keys(
+                        function (keys) {
+                            //Check if OutSystems local key exists
+                            if(keys.indexOf(LOCAL_STORAGE_KEY) >= 0) {
+                                // If succeded, attempt to get OutSystems local key
+                                ss.get(
+                                    function (value) {
+                                        lskCache = value;
+                                        successCallback(lskCache);
+                                    },
+                                    function (error) {
+                                        OutSystemsNative.Logger.logError("Error getting local storage key from keychain: " + error, "SecureSQLiteBundle");
```

Comment:
`OutSystemsNative.Logger` is a non explicit dependency from this plugin, provided that eventually this plugin is only used with OutSystems applications, it shouldn't be a problem, however, this repository is public and also, plugin.xml doesn't state any dependencies with the Logger, which would cause runtime errors when using this plugin.

Reply:
Although this repository is public was entirely created by OutSystems. Given that it is no fork I do not see the problem in referencing OutSystems assets in an OutSystems context.

Perhaps the discussion should be instead whether or not this repository should remain public.

Reply:
Public or not, it's an OutSystems plugin and it should properly reference its dependencies.

Reply:
And, by the way, the OutSystemsNative object must be declared to ensure the dependency load.